react/week4/git-users/src/UserContext.js

```diff
@@ -1,37 +1,60 @@
 import React, { useState, useEffect, createContext } from "react";
+import axios from "axios";
 
 export const userContext = createContext();
 
-const UserProvider = (props) => {
+const UserProvider = ({ children }) => {
   const [user, setUser] = useState([]);
   const [loading, setLoading] = useState(false);
   //   const [hasResult, setHasResult] = useState(true);
   const getUsers = async (elm) => {
-    // setLoading(true);
-    const url = `https://api.github.com/search/users?q=${elm}`;
-    const Api = await fetch(url)
-      .then((response) => {
-        return response.json();
-      })
-      .then((data) => {
-        const { items } = data;
-        let userList = [];
-        items.forEach((item) => {
-          const { login } = item;
-          userList.push(login);
-        });
-        if (data.total_count > 0) {
-          setUser(userList);
-        } else {
-          setUser(["No Results!"]);
-        }
-
-        setLoading(false);
-        console.log(data);
-      })
-      .catch((error) => {
-        console.log("error");
+    setLoading(true);
+    try {
+      const userData = await axios.get(
+        `https://api.github.com/search/users?q=${elm}`
+      );
+      const {
+        data: { items },
+        data: { total_count },
+      } = await userData;
+      let userList = [];
+      items.forEach((item) => {
+        const { login } = item;
+        userList.push(login);
       });
```

Comment:
```
const userList = items.map(item => item.login);
```

does exactly what you do, but cleaner. Avoid using `let` unless you really need to mutate the data afterward. In your case, you don't because you display them on UI without further process them.